### PR TITLE
[onert] Introduce LossReductionType

### PR DESCRIPTION
**Pull request by jyoungyun**
This commit introduces LossReductionType for training feature. It will use SumOverBatchSize reduction type, if the reduction type is not specified.

ONE-DCO-1.0-Signed-off-by: Jiyoung Yun <jy910.yun@samsung.com>

### CODE REVIEW

File: runtime/onert/api/include/nnfw_experimental.h [hunk 1]
```diff
@@ -197,8 +213,9 @@ typedef struct nnfw_train_info
   float learning_rate = 0.001f;
   /** Batch size */
   uint32_t batch_size = 1;
-  /** loss type */
-  NNFW_TRAIN_LOSS loss = NNFW_TRAIN_LOSS_MEAN_SQUARED_ERROR;
+  /** loss info */
+  nnfw_loss_info loss_info{.loss = NNFW_TRAIN_LOSS_MEAN_SQUARED_ERROR,
```
Comment (jyoungyun) [inline runtime/onert/api/include/nnfw_experimental.h:217]: It should be an `INVALID` loss type. But that is not the scope of this PR. I will make another pr for this.

**Comment by glistening** [comment]
@jyoungyun As I read from https://www.tensorflow.org/api_docs/python/tf/keras/losses/Reduction, `SUM_OVER_BATCH_SIZE` seems to be natural choice for almost cases. Is there an occasion to use other loss reduction type?

File: runtime/onert/api/include/nnfw_experimental.h [hunk 2]
```diff
@@ -180,12 +180,28 @@ typedef enum
...
 } NNFW_TRAIN_LOSS;
 
+typedef enum
+{
+  /** Invalid */
+  NNFW_TRAIN_LOSS_REDUCTION_INVALID = 0,
+  /** Scalar sum divided by number of elements in losses */
+  NNFW_TRAIN_LOSS_REDUCTION_SUM_OVER_BATCH_SIZE = 1,
+  /** Scalar sum of weighted losses */
+  NNFW_TRAIN_LOSS_REDUCTION_SUM = 2,
```
[the remark this answers is not in the document]
Comment (jyoungyun) [reply runtime/onert/api/include/nnfw_experimental.h:190]: Yes, we need to implement `NONE` type. However, there is currently no plan to implement `NONE` type. So I didn't add it. Do you think we should add `NONE` type even if don't implement this now?

**Review by glistening — APPROVED** [review]
LGTM

File: runtime/onert/api/include/nnfw_experimental.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (ragmani) [reply runtime/onert/api/include/nnfw_experimental.h:190]: I thought `NONE` is the default type for loss reduction. If not, you don't need to add `NONE` type. If the type is default for loss reduction, I think it's fine to add `NONE` type regardless implementing the type. It's not required if you don't want to.
By the way, onert supports `SUM_OVER_BATCH_SIZE` now?

**Review by ragmani — APPROVED** [review]
LGTM

File: runtime/onert/api/include/nnfw_experimental.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (jyoungyun) [reply runtime/onert/api/include/nnfw_experimental.h:190]: @ragmani 

> By the way, onert supports `SUM_OVER_BATCH_SIZE` now?

Yes, the `SUM_OVER_BATCH_SIZE` is already implemented and I'm going to implement `SUM`. So I added those two types. :)
